### ISSUE TITLE
Inconsitent behaviour of asadamin set property commands for CertificateRealm

### ISSUE THREAD

**Issue by borisheithecker** [payara/Payara#4611]
# Description #
----------

Configuring a certificate realm via asadmin commands show inconsistent behaviour. 

## Current and expected outcome, steps to reproduce

Caller of resources secured with client certificate authentication are not always assigned to designated roles. Please, see this repository with code and steps to reproduce: 
https://github.com/borisheithecker/payara-clientcert-test3

The reason seems to be that on changes in the server configuration, ```CertificateRealm.init(Properties)``` is called in an inconsistent manner. This seems to be related to the notification mechanism in [Transactions.java](https://github.com/payara/Payara/blob/master/nucleus/hk2/hk2-config/src/main/java/org/jvnet/hk2/config/Transactions.java) at ```Transactions.ConfigListenerNotifier.prepare(...)```

https://github.com/payara/Payara/issues/4596 and https://github.com/payara/Payara/issues/4533 are possibly related.

## Environment ##

- **Payara Version**: 5.201
- **Edition**: 
- **JDK Version**: openjdk 11.0.6 2020-01-14 / as in payara/server-full:5.201
- **Operating System**:

**Comment by fturizo** [comment]
Thanks for this report, @borisheithecker. I've been able to reproduce the problem you are encountering, so I'm investigating the issue in detail before escalating it to our engineering team.

**Comment by MeroRai** [comment]
Hi @borisheithecker,

I have could reproduce the issue in our current release of Payara community release, which is  `5.2020.6`. I have now escalated it to our engineering team and created an internal issue `FISH-787` to track it. Thank you for your assist.

**Comment by borisheithecker** [comment]
Yes, it's exactly reproducible in Payara 5.2020.6. I've updated the github respository.
HTTP/2 should be disabled for client certificate authentication.
